base_layer/core/src/mempool/utx_pool/utx_pool_storage.rs

```diff
@@ -0,0 +1,165 @@
+    /// reached and the new transaction has a higher priority than the currently stored lowest priority transaction.
+    pub fn insert_utx(&mut self, utx: Transaction) -> Result<(), UTxPoolError> {
+        if utx.body.kernels.len() == 0 {
+            return Err(UTxPoolError::NoKernelsFound);
+        }
+        let utx_key = utx.body.kernels[0].excess_sig.clone();
+        if !self.utx_hmap.contains_key(&utx_key) {
+            let prioritized_utx = PrioritizedTransaction::try_from(utx)?;
+            if self.utx_hmap.len() >= self.config.storage_capacity {
+                if prioritized_utx.priority < *self.lowest_priority() {
+                    return Ok(());
+                }
+                self.remove_lowest_priority_utx();
+            }
+
+            self.utx_tree.insert(prioritized_utx.priority.clone(), utx_key.clone());
+            self.utx_hmap.insert(utx_key, prioritized_utx);
+        }
+        Ok(())
+    }
+
+    /// Insert a set of new transactions into the UTxPoolStorage
+    pub fn insert(&mut self, utxs: Vec<Transaction>) -> Result<(), UTxPoolError> {
+        for utx in utxs.into_iter() {
+            self.insert_utx(utx)?;
+        }
+        Ok(())
+    }
+
+    /// Check if a transaction is stored in the UTxPoolStorage
+    pub fn contains_key(&self, tx_excess_sig: &Signature) -> bool {
+        self.utx_hmap.contains_key(tx_excess_sig)
+    }
+
+    /// Returns a set of the highest priority unconfirmed transactions, that can be included in a block
+    pub fn highest_priority_utxs(&self, n: usize) -> Result<Vec<Transaction>, UTxPoolError> {
+        let mut selected_utxs: Vec<Transaction> = Vec::with_capacity(n);
+        for (_, utx_key) in self.utx_tree.iter().rev() {
+            selected_utxs.push(
+                self.utx_hmap
+                    .get(utx_key)
+                    .ok_or(UTxPoolError::StorageOutofSync)?
+                    .transaction
+                    .clone(),
+            );
+            if selected_utxs.len() >= n {
+                break;
+            }
+        }
+        Ok(selected_utxs)
+    }
+
+    /// Remove all published transactions from the UTxPoolStorage
+    pub fn remove_published(&mut self, published_block: &Block) -> Vec<Transaction> {
+        let mut removed_txs: Vec<Transaction> = Vec::new();
+        published_block.body.kernels.iter().for_each(|kernel| {
+            if let Some(ptx) = self.utx_hmap.get(&kernel.excess_sig) {
+                self.utx_tree.remove(&ptx.priority);
+                removed_txs.push(self.utx_hmap.remove(&kernel.excess_sig).unwrap().transaction);
+            }
+        });
+        removed_txs
+    }
+
+    /// Remove orphaned transactions from the UTxPoolStorage. These transactions were orphaned by the provided published
+    /// block.
+    pub fn remove_orphaned(&mut self, published_block: &Block) -> Vec<Transaction> {
```

Comment:
```suggestion
    pub fn remove_double_spends(&mut self, published_block: &Block) -> Vec<Transaction> {
```

Reply:
This functions behaviour was changed to discard double spends, was renamed to `discard_double_spends`